src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -60,7 +60,7 @@ public static partial class JsonSerializer
                     }
                     else if (writer.CurrentDepth >= options.EffectiveMaxDepth)
                     {
-                        ThrowHelper.ThrowJsonException_DepthTooLarge(writer.CurrentDepth, state, options);
+                        ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(options.MaxDepth);
```

Comment:
Should this be `options.EffectiveMaxDepth`? It was, previously.

Reply:
It doesn't matter. I changed it to the public property.